### PR TITLE
Rails 5.0 Support

### PR DESCRIPTION
**Pull request by etagwerker**
Hey @sauloperez @wyhaines, 

This PR takes @wyhaines's work (https://github.com/redbooth/departure/pull/81) and adds a couple of changes to make the gem Rails 5.0 compatible.

I solved a couple of conflicts in the process.

Please let me know if you have any comments.

Thank you!

### CODE REVIEW

**Comment by wyhaines** [comment]
Nice. Thanks for the Rails 5.0 compatibility and fixes.

**Comment by guilleiguaran** [comment]
Great!! 

Does this works for 5.1 as well or just for 5.0?

**Comment by etagwerker** [comment]
@wyhaines Thank you for doing most of the work! 👍 

@guilleiguaran I think so, I just applied my last commit to this PR https://github.com/departurerb/departure/pull/14 and it seems to work. 

> Edit: I guess this should be merged into master and then branch v5.0 should be created from master, then a new PR should follow changing the dependency to ~> 5.1.4 and a new v5.1 branch should be created.

I like this idea.

**Comment by etagwerker** [comment]
@sauloperez @wyhaines I just fixed the build. This is now passing. 

Please let me know if you see anything wrong with it.

**Comment by sauloperez** [comment]
Welcome @etagwerker and thanks a lot for your awesome contribution! it was very much needed!

**Comment by wyhaines** [comment]
It is looking very good to me.  Thank you.

File: spec/integration_spec.rb [hunk 1]
```diff
@@ -16,33 +16,33 @@ class Comment < ActiveRecord::Base; end
...
+        ActiveRecord::Migration.verbose = true
         example.run
         ActiveRecord::Migration.verbose = original_verbose
       end
 
       it 'sends the output to the stdout' do
         expect do
           ActiveRecord::Migrator.new(direction, migration_fixtures, 1).migrate
-        end.to_not output.to_stdout
+        end.to output.to_stdout
```
Comment (sauloperez) [inline spec/integration_spec.rb]: what does exactly happen with these context? I'm not sure who but someone had also a problem with them. We should create an issue for it.

File: lib/departure/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Departure
-  VERSION = '4.0.1'.freeze
+  VERSION = '5.0.1'.freeze
```
Comment (sauloperez) [inline lib/departure/version.rb]: can you please remove this change as you did with #14 ?

**Comment by etagwerker** [comment]
@sauloperez Just pushed a couple of commits addressing your comments. Thanks for the quick response! 🎉

**Comment by wyhaines** [comment]
@etagwerker Thank you.

**Comment by sauloperez** [comment]
following @guilleiguaran comment, I just pushed the `v5.0` branch. All bug fixes for Rails 5.0.X should branch off from it.

**Pull request merged**